### PR TITLE
Testing 2-3-24 last tested version

### PR DESCRIPTION
**Pull request by 3LucasZ**
Saturday tuning/testing + constants clean up + fix merge conflicts

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 1]
```diff
@@ -38,27 +36,28 @@ public class Shooter extends SubsystemBase implements Loggable {
 
   private TalonFX shooterMotorFollower;
   private TalonFX feederMotor;
+
   private TalonFXSimState shooterMotorSim;
   private FlywheelSim flywheelSimModel;
+
   final VelocityVoltage velocityShooterRequest =
-      new VelocityVoltage(0).withEnableFOC(true).withSlot(0);
+      new VelocityVoltage(0).withEnableFOC(false).withSlot(0).withFeedForward(0);
```
Comment (ThatXliner) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java:44]: No FOC?

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 2]
```diff
@@ -38,27 +36,28 @@ public class Shooter extends SubsystemBase implements Loggable {
...
   final VelocityVoltage velocityFeederRequest =
-      new VelocityVoltage(0).withEnableFOC(true).withSlot(1);
+      new VelocityVoltage(0).withEnableFOC(false).withSlot(0).withFeedForward(0);
```
Comment (ThatXliner) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java:46]: u sure changing the slot to 0 will work?. Feedforward should be put into a constant.

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 3]
```diff
@@ -38,27 +36,28 @@ public class Shooter extends SubsystemBase implements Loggable {
...
+
   final MotionMagicVelocityVoltage motionMagicShooterRequest =
-      new MotionMagicVelocityVoltage(0).withEnableFOC(true).withSlot(0);
+      new MotionMagicVelocityVoltage(0).withEnableFOC(false).withSlot(0);
   final MotionMagicVelocityVoltage motionMagicFeederRequest =
-      new MotionMagicVelocityVoltage(0).withEnableFOC(true).withSlot(1);
+      new MotionMagicVelocityVoltage(0).withEnableFOC(false).withSlot(0);
   final MotionMagicVelocityVoltage motionMagicShooterFollowerRequest =
-      new MotionMagicVelocityVoltage(0).withEnableFOC(true).withSlot(2);
+      new MotionMagicVelocityVoltage(0).withEnableFOC(false).withSlot(0);
```
Comment (ThatXliner) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java:55]: we can def refactor this to reduce code dupe (and prob will be worth it too). Like some sort of factory function.

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 4]
```diff
@@ -78,12 +77,14 @@ private void configureRealHardware() {
     shooterMotor = new TalonFX(ShooterConstants.kShooterMotorID);
     shooterMotorFollower = new TalonFX(ShooterConstants.kShooterMotorFollowerID);
     feederMotor = new TalonFX(ShooterConstants.kFeederMotorID);
-    shooterMotor.setInverted(false);
+
+    shooterMotor.setInverted(true);
```
Comment (ThatXliner) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java:81]: this was from our in-person testing, correct?

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 5]
```diff
@@ -167,28 +169,29 @@ public void setInputFeederVoltage(double voltage) {
...
       if (kDebugEnabled) {
         System.out.println("Shooter velocity set to: " + shooterRps);
         System.out.println("Shooter follower velocity set to: " + followerRps);
       }
     } else {
-      shooterMotor.setControl(velocityShooterRequest.withVelocity(shooterRps));
-      shooterMotorFollower.setControl(motionMagicShooterFollowerRequest.withVelocity(followerRps));
+      shooterMotor.setControl(velocityShooterRequest.withVelocity(shooterRps).withFeedForward(5));
+      shooterMotorFollower.setControl(
+          motionMagicShooterFollowerRequest.withVelocity(followerRps).withFeedForward(5));
```
Comment (ThatXliner) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java:182]: Constant.

File: src/main/java/frc/robot/subsystems/swerve/SwerveModule.java [hunk 6]
```diff
@@ -48,16 +48,16 @@ public SwerveModule(int moduleNumber, SwerveModuleConstants moduleConstants) {
     this.angleOffset = moduleConstants.angleOffset;
 
     /* Angle Encoder Config */
-    angleEncoder = new CANcoder(moduleConstants.cancoderID);
+    angleEncoder = new CANcoder(moduleConstants.cancoderID, "mani");
```
Comment (ThatXliner) [inline src/main/java/frc/robot/subsystems/swerve/SwerveModule.java:51]: why?

File: src/main/java/frc/robot/subsystems/swerve/SwerveModule.java [hunk 7]
```diff
@@ -48,16 +48,16 @@ public SwerveModule(int moduleNumber, SwerveModuleConstants moduleConstants) {
...
 
     /* Angle Motor Config */
-    mAngleMotor = new TalonFX(moduleConstants.angleMotorID);
+    mAngleMotor = new TalonFX(moduleConstants.angleMotorID, "mani");
     mAngleMotor.getConfigurator().apply(Robot.ctreConfigs.swerveAngleFXConfig);
     resetToAbsolute();
 
     /* Drive Motor Config */
-    mDriveMotor = new TalonFX(moduleConstants.driveMotorID);
+    mDriveMotor = new TalonFX(moduleConstants.driveMotorID, "mani");
```
Comment (realjoshuau) [inline src/main/java/frc/robot/subsystems/swerve/SwerveModule.java:60]: todo: pull CanBus out to a constant. Not needed for now

**Review by realjoshuau — APPROVED** [review]
LGTM; couple of changes regarding logging that I manually edited.

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java (hunk 1 above)
Comment (3LucasZ) [reply src/main/java/frc/robot/subsystems/shooter/Shooter.java:44]: We will play around with it when further tuning the shooter later this week.

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java (hunk 3 above)
Comment (3LucasZ) [reply src/main/java/frc/robot/subsystems/shooter/Shooter.java:55]: We will in a future branch/commit. This is the last version tested irl, can't make too big of changes.

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java (hunk 4 above)
Comment (3LucasZ) [reply src/main/java/frc/robot/subsystems/shooter/Shooter.java:81]: Yes

File: src/main/java/frc/robot/subsystems/intake/commands/IntakeSetVoltage.java [hunk 8]
```diff
@@ -22,11 +22,11 @@ public IntakeSetVoltage(Intake intake, double volts) {
...
   @Override
   public void initialize() {
-    intakeSubsystem.setInputVoltage(volts);
+    intakeSubsystem.setOutputVoltage(volts);
   }
 
   @Override
   public boolean isFinished() {
-    return false;
+    return true;
```
[the remark this answers is not in the document]
Comment (3LucasZ) [reply src/main/java/frc/robot/subsystems/intake/commands/IntakeSetVoltage.java:30]: Yes, this is intentional for the voltage to keep running forever, just for testing. I doubt it causes the motor to stall, but we will further test in a new branch.

File: src/main/java/frc/robot/subsystems/swerve/SwerveModule.java (hunk 7 above)
Comment (3LucasZ) [reply src/main/java/frc/robot/subsystems/swerve/SwerveModule.java:60]: Ok, added TODO

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java (hunk 2 above)
Comment (3LucasZ) [reply src/main/java/frc/robot/subsystems/shooter/Shooter.java]: init is at zero, we change the value when using the request later on. No need to change this 0 into constant. Rn using slot 0.

**Pull request merged**